rollbar/rollbar-php-wordpress
# Fix unit tests and PHP8.1+ compatability

## Description of the change

Fix #113 and #102

This updates the unit tests to use the [Yoast Pollyfill](https://github.com/Yoast/PHPUnit-Polyfills) and also [wp-env](https://www.npmjs.com/package/@wordpress/env) for testing across multiple versions of PHP. 

The underlying [rollbar-php](https://github.com/rollbar/rollbar-php) is both updated and kept the same. It's updated for sites running PHP8+, but kept the same for PHP7.  This introduces a bit of technical debt as the actual dependencies aren't in the composer file but instead are in the `php7` and `php8` folder. To help manage that, a script is included for updating those dependencies. 

There are a few spots in the code that now also include PHP version specific code. When php7 support is dropped, these will need to be cleaned up. 

Additionally, github actions are added in order to allow unit tests to run across multiple versions of PHP

## Testing Instructions

If you want to test this change manually, you can use wp-env. clone the repo and make sure git, docker, and node are installed on your machine.  Then run 

`npm install`
`WP_ENV_PHP_VERSION="7.0" npm run wp-env start -- --update` to start with PHP 7.0

You can login to Wordpress at http://localhost:8888/ with the username `admin` and the password `password`

If you want to test a different version of PHP, you can restart the env with a different PHP version passed in.  For example, `WP_ENV_PHP_VERSION="8.2" npm run wp-env start -- --update` to start with PHP 8.2

## Type of change

- [ ] New release

## Checklists

### Development


### Code review

- [ ] "Ready for review" label attached to the PR and reviewers assigned
- [ ] Issue from task tracker has a link to this pull request
- [ ] Changes have been reviewed by at least one other engineer
